presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetTypeUtils.java

```diff
@@ -193,6 +198,40 @@ public static ParquetEncoding getParquetEncoding(Encoding encoding)
             return Optional.empty();
         }
         DecimalMetadata decimalMetadata = descriptor.getPrimitiveType().getDecimalMetadata();
-        return Optional.of(DecimalType.createDecimalType(decimalMetadata.getPrecision(), decimalMetadata.getScale()));
+        return Optional.of(createDecimalType(decimalMetadata));
+    }
+
+    private static Type createDecimalType(DecimalMetadata decimalMetadata)
+    {
+        return DecimalType.createDecimalType(decimalMetadata.getPrecision(), decimalMetadata.getScale());
+    }
+
+    private static Type getInt32Type(RichColumnDescriptor descriptor)
+    {
+        OriginalType originalType = descriptor.getPrimitiveType().getOriginalType();
+        if (originalType == DECIMAL) {
```

Comment:
Please use switch case for the if/else chain here and in `getInt64Type` below.

Reply:
If I remember correctly I changed this from a switch because originalType is nullable (I forget exactly how but I ran into this) so I changed it to just if/else because I couldn't switch on a null value, could just short circuit the null check though then do a switch. 

I'll pull out the `TIMESTAMP` changes to move forward with just `DATE` changes.

Reply:
Updated to use `switch` but a null check needs to take place first then on `originalType` since it's nullable